src/main/java/uk/gov/companieshouse/insolvency/data/service/InsolvencyServiceImpl.java

```diff
@@ -148,17 +142,17 @@ public void deleteInsolvency(String contextId, String companyNumber) {
                         companyNumber));
             }
 
-            insolvencyRepository.deleteById(companyNumber);
-            logger.info(String.format(
-                    "Company insolvency is deleted in MongoDB with "
-                            + "context id %s and company number %s",
+            insolvencyApiService.invokeChsKafkaApi(contextId, insolvencyDocumentOptional.get(),
```

Comment:
Please review this implementation following what is described in the story as per below:

If its the delete flow

Fetch document from DB for a given company number, if not exists thats the end of flow, no need to call chs-kafka-api

Call chs-kafka-api only if entry exists in DB 

Delete from DB only if the above call to chs-kafka-api was successful (2xx series response).

Reply:
Doesnt this do what the story requires:
<img width="772" alt="image" src="https://user-images.githubusercontent.com/16381559/171160611-f57dca80-6117-47ef-afcd-cb36ce8070e7.png">
If document is empty then doesnt carry on the flow, if it exists it will call Chs Kafka API, which then if throws non 200 will throw exception ending the flow